### PR TITLE
[WIP] Add OpenIDConnect Authentication

### PR DESCRIPTION
**Pull request by aalexanderr**
This PR aims to add OIDC integration for scancode.io.

Why:
We want to run scancode.io on public address and would prefer to have the ability to give access to it only to the ONAP community.

I've asked question on weekly meeting if you guys are interested in OIDC integration for scancode.io (as we were adding it for our instance) and got response that yes, you are. I probably should've created an issue for it.

Commits currently pushed were authored by Mateusz Sobocinski (@matteuszs @msobocinsk) who just left our company thus I'll  try to adjust this to be mergable. I'll squash the commits, add Mateusz's SoB & add docs regarding usage.

We have tested it with keycloak (there is new target in the makefile adding keycloak instance in docker but needs manual configuration for now, should I describe it here/automate it so it can be checked easily?)

The main question I have before removing WIP label is - should it be enabled by default?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,7 +116,7 @@ sqlite:
 	@$(MAKE) migrate
 
 run:
-	${MANAGE} runserver 8001
+	${MANAGE} runserver 0.0.0.0:8001
```
Comment (aalexanderr) [inline Makefile:119]: should probably be reverted, i see no need to expose on all addresses

File: Makefile [hunk 2]
```diff
@@ -129,4 +137,21 @@ docs:
...
 	@${ACTIVATE} sphinx-build docs/ docs/_build/
 
+keycloak:
+	@echo "-> Run Keycloak Identity provider"
+	docker run -dit --name scancodeio-keycloak \
+                    --env KEYCLOAK_USER=${KEYCLOAK_USER} \
+                    --env KEYCLOAK_PASSWORD=${KEYCLOAK_PASSWORD} \
+                    --publish ${KEYCLOAK_HTTPS_BIND_PORT}:8443 \
+                    jboss/keycloak
+	@echo "-> Keycloak launched. Please visit https://localhost:${KEYCLOAK_HTTPS_BIND_PORT} and configure realm"
```
Comment (aalexanderr) [inline Makefile:147]: add script to configure using kcadm.sh
https://github.com/keycloak/keycloak-documentation/blob/master/server_admin/topics/admin-cli.adoc

File: Makefile [hunk 3]
```diff
@@ -129,4 +137,21 @@ docs:
 	rm -rf docs/_build/
 	@${ACTIVATE} sphinx-build docs/ docs/_build/
 
+keycloak:
+	@echo "-> Run Keycloak Identity provider"
+	docker run -dit --name scancodeio-keycloak \
```
Comment (aalexanderr) [inline Makefile:142]: My guess is that -it flags are not needed here

File: Makefile [hunk 4]
```diff
@@ -129,4 +137,21 @@ docs:
...
+                    --env KEYCLOAK_USER=${KEYCLOAK_USER} \
+                    --env KEYCLOAK_PASSWORD=${KEYCLOAK_PASSWORD} \
+                    --publish ${KEYCLOAK_HTTPS_BIND_PORT}:8443 \
+                    jboss/keycloak
+	@echo "-> Keycloak launched. Please visit https://localhost:${KEYCLOAK_HTTPS_BIND_PORT} and configure realm"
+
+oidc:
+	@echo "-> Configure OIDC settings"
+	@echo "" >> ${ENV_FILE}
+	@echo OIDC_OP_REALM_ENDPOINT=\"https://keycloak:${KEYCLOAK_HTTPS_BIND_PORT}/auth/realms/${KEYCLOAK_REALM_NAME}/\" >> ${ENV_FILE}
```
Comment (aalexanderr) [inline Makefile:152]: add env for `https://keycloak` so it can be easily configured when having keycloak running on different domain

**Comment by tdruez** [comment]
Could you provide some context and explanation about those changes?

**Comment by aalexanderr** [comment]
> Could you provide some context and explanation about those changes?

Added some context to the PR msg, let me know if I missed something and I'll be happy to write it down :)

**Pull request closed without merging**